include/mvd/mvd_generic.hpp

```diff
@@ -57,25 +56,31 @@ inline MVDFile* open_mvd(const std::string & filename) {
 /// performed, but no guarantees about file correctness are made.
 ///
 /// \param filename the path of the file to open
+/// \param extra_arg extra argument to open tsv file or sonata parameter
 /// \return a shared pointer to a mvd::File object
 ///
 inline std::shared_ptr<File> open(const std::string& filename,
-                                  const std::string& population="default") {
+                                  const std::string& extra_arg="default") {
     std::shared_ptr<File> mvdfile;
     switch(_mvd_format(filename)) {
         case MVDType::MVD2:
             throw std::runtime_error("MVD2 not supported in the new MVD API");
         case MVDType::MVD3:
             try {
-                mvdfile.reset(new MVD3::MVD3File(filename));
+                if (_mvd_format(extra_arg) == MVDType::TSV) {
```

Comment:
As per the previous comment, maybe could be changed.

Reply:
It is removed completely now, so that the `tsv file` can be read with a `readTSVInfo()` function